src/wlapplication_mousewheel_options.cc

```diff
@@ -0,0 +1,323 @@
+/*
+ * Copyright (C) 2021 by the Widelands Development Team
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ */
+
+#include "wlapplication_mousewheel_options.h"
+
+#include <cassert>
+#include <cstdlib>
+#include <map>
+
+#include "wlapplication_options.h"
+
+static bool mousewheel_use_2d_defaults_cache = false;
+
+enum class MousewheelOptionType { kBool, kKeymod };
+
+struct MousewheelOption {
+	const std::string internal_name_;
+	const MousewheelOptionType type_;
+	const uint16_t default_;
+	const uint16_t default_2D_;
+
+	MousewheelOption(std::string n, MousewheelOptionType t, uint16_t def, uint16_t def2D)
+	   : internal_name_(n), type_(t), default_(def), default_2D_(def2D) {
+	}
+
+	bool get_bool(bool use_2d_default) const {
+		assert(type_ == MousewheelOptionType::kBool);
+		const bool def = static_cast<bool>(use_2d_default ? default_2D_ : default_);
```

Comment:
Actually the cast is not necessary – just use `!= 0` for readability
```c++
 		const bool def = (use_2d_default ? default_2D_ : default_) != 0;
```

Reply:
This is to signify that these are actually bool values, and only stored in uint-s so that the same struct can be used for keymods and bools, so they can be kept in the same map, which is useful to keep the related keymod and bool settings next to each other in the config file. So we're just converting them back to bool here.

Does this make sense?

Reply:
https://docs.microsoft.com/en-us/previous-versions/visualstudio/visual-studio-2010/b6801kcy(v=vs.100)?redirectedfrom=MSDN
> 'type' : forcing value to bool 'true' or 'false' (performance warning)
>
> This warning is generated when a value that is not bool is assigned or coerced into type bool. Typically, this message is caused by assigning int variables to bool variables where the int variable contains only values true and false, and could be redeclared as type bool. If you cannot rewrite the expression to use type bool, then you can add "!=0" to the expression, which gives the expression type bool. Casting the expression to type bool will not disable the warning, which is by design.